pkg/catalogv2/helm/helm3.go

```diff
@@ -121,16 +121,18 @@ func decodeHelm3(data string) (*release.Release, error) {
 	// For backwards compatibility with releases that were stored before
 	// compression was introduced we skip decompression if the
 	// gzip magic header is not found
-	if bytes.Equal(b[0:3], magicGzip) {
-		r, err := gzip.NewReader(bytes.NewReader(b))
-		if err != nil {
-			return nil, err
-		}
-		b2, err := ioutil.ReadAll(r)
-		if err != nil {
-			return nil, err
+	if len(b) > 2 {
+		if bytes.Equal(b[0:3], magicGzip) {
+			r, err := gzip.NewReader(bytes.NewReader(b))
+			if err != nil {
+				return nil, err
+			}
+			b2, err := ioutil.ReadAll(r)
+			if err != nil {
+				return nil, err
+			}
+			b = b2
```

Comment:
```suggestion
	if len(b) <= 3 {
	    // not a Helm 3 release
	    return nil, nil
	}
	if bytes.Equal(b[0:3], magicGzip) {
		r, err := gzip.NewReader(bytes.NewReader(b))
		if err != nil {
			return nil, err
		}
		b2, err := ioutil.ReadAll(r)
		if err != nil {
			return nil, err
		}
		b = b2
```

Reply:
nit: can we just add the check above instead of putting everything in the `if` block?

Is there any reason why we should do the `json.Unmarshal` if `len(b) <= 3`?

Reply:
No reason. it will error out and the error will be ignored. 

https://github.com/rancher/rancher/blob/release/v2.7/pkg/controllers/dashboard/helm/apps.go#L175